### PR TITLE
Prune Blacklisted Block Hashes Past Finalized State

### LINKED ISSUE #2298: Prune Blacklisted Block Hashes Past FinalizedState
As a follow-up to #2297, we should prune the map of blacklisted block hashes once we have a new finalized state. At that point, all block hashes that came before the new finalized state can be safely deleted, as those blocks will never be considered again.

### PR DESCRIPTION
**Pull request by JerryFireman**
TODO(you): choose "part of" or "resolves" and the associated github issue #.

Resolves #2298 

---

# Description

**Write why you are making the changes in this pull request**
Once we have a new finalized state, blacklisted block hashes can be safely deleted

**Write a summary of the changes you are making**
Update beacon-chain/db/state.go to remove blacklisted block hashes

**Link anything that would be helpful or relevant to the reviewers**

### CODE REVIEW

**Comment by JerryFireman** [comment]
@terencechain ok will do

File: beacon-chain/db/state.go [hunk 1]
```diff
@@ -191,11 +191,18 @@ func (db *BeaconDB) SaveJustifiedState(beaconState *pb.BeaconState) error {
 
 // SaveFinalizedState saves the last finalized state in the db.
 func (db *BeaconDB) SaveFinalizedState(beaconState *pb.BeaconState) error {
+	// Delete blacklisted block hashes which came before the new finalized state
+	defer func() {
```
Comment (prestonvanloon) [inline beacon-chain/db/state.go]: No need to create a new goroutine here. Please remove the `defer func() { ... }()`

**Comment by JerryFireman** [comment]
@prestonvanloon @terencechain just updated PR to add test and eliminate extraneous go routine

**Comment by JerryFireman** [comment]
@prestonvanloon @terencechain Just checking to see if you have any feedback for me?

**Comment by JerryFireman** [comment]
@prestonvanloon @terencechain Just checking to see if you have any feedback for me?

**Comment by JerryFireman** [comment]
@terencechain @prestonvanloon This pull request was by stale botclosed due to inactivity. I am guessing that you were unable to review because you were dealing with testnet issues.

**Pull request closed without merging**